### PR TITLE
Link PyNEST functions in documentaton

### LINKED ISSUE #1878: Automatically link PyNEST functions
Currently, the absence of automatically linked PyNEST functions leads to redundancy and repetitive descriptions.

As @jougs suggested in stinebuu/nest-simulator#8, it would be good to have an automatic linking of all PyNEST functions that are mentioned throughout the documentation (e.g. ``nest.Connect()``) to the corresponding anchor in the PyNEST API documentation.

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR provides links to the API functions in the documentation, specifically restructuredtext files in doc/userdoc.
There are still files to update in the nest-3 guides, but waiting for #1929  to be merged, so this is still incomplete.

Note: set() and get() methods are in both SynapseCollection and NodeCollection class, so the class name needs to be included in the link.  (This is mentioned in styleguide  #1986 )

Resolves #1878

### CODE REVIEW

**Comment by jougs** [comment]
@steffengraber, @jasperalbers: could you please review? If you don't have time right now, please quickly indicate this, so we can try to find other reviewers. Thanks!

File: doc/userdoc/guides/connection_management.rst [hunk 1]
```diff
@@ -130,12 +130,12 @@ more details on this interface, see the Git repository of `libneurosim
 In contrast to the other rules for creating connections, this rule
 relies on a Connection Generator object to describe the connectivity
 pattern in a library-specific way. The Connection Generator is handed
-to ``Connect()`` under the key `cg` of the connection specification
+to :py:func:`.Connect` under the key `cg` of the connection specification
 dictionary and evaluated internally. If the Connection Generator
 provides values for connection weights and delays, their respective
 indices can be specified under the key `params_map`. Alternatively,
```
Comment (jasperalbers) [inline doc/userdoc/guides/connection_management.rst:136]:
```suggestion
to :py:func:`.Connect` under the key ``cg`` of the connection specification
dictionary and evaluated internally. If the Connection Generator
provides values for connection weights and delays, their respective
indices can be specified under the key ``params_map``. Alternatively,
```

Should these be double apostrophes as many other code-like objects or are the single ones intended here?

File: doc/userdoc/guides/connection_management.rst [hunk 2]
```diff
@@ -130,12 +130,12 @@ more details on this interface, see the Git repository of `libneurosim
...
 all synapse parameters can be specified using the synapse
-specification argument to ``Connect()``
+specification argument to ``Connect1()``.
```
Comment (jasperalbers) [inline doc/userdoc/guides/connection_management.rst:138]:
```suggestion
specification argument to ``Connect()``.
```
possible typo

File: doc/userdoc/guides/spatial/guide_spatially_structured_networks.rst [hunk 3]
```diff
@@ -877,7 +877,7 @@ Probabilistic connections between layers can be generated in two different
 ways:
 
 Free probabilistic connections using `pairwise_bernoulli`
```
Comment (jasperalbers) [inline doc/userdoc/guides/spatial/guide_spatially_structured_networks.rst:879]:
```suggestion
Free probabilistic connections using ``pairwise_bernoulli``
```
Possible miss of using double quotes (ignore this if intended).

**Review by jasperalbers — COMMENTED** [review]
Works as intended, very useful! I left a few comments.

File: doc/userdoc/guides/connection_management.rst [hunk 4]
```diff
@@ -18,13 +18,13 @@ target nodes that will be connected in an all-to-all fashion.
 Each call to the function will establish the connectivity between pre-
 and post-synaptic populations according to a certain pattern or
 rule. The desired pattern is specified by simply stating the rule name
-as third argument to ``Connect()``, or by setting the key `rule` in
-the `connectivity specification` dictionary ``conn_spec`` alongside
+as third argument to ``Connect()``, or by setting the key ``rule`` in
```
[the remark this answers is not in the document]
Comment (jessica-mitchell) [reply doc/userdoc/guides/connection_management.rst:21]: Yes, I was trying to only link each function once per paragraph or section.

**Comment by jessica-mitchell** [comment]
Hi @jasperalbers I have addressed your comments, could you please take another look? thanks!

**Review by steffengraber — CHANGES_REQUESTED** [review]
Excellent! Really helpful! Found only one little thing.

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_handling_connections.rst [hunk 5]
```diff
@@ -203,17 +203,17 @@ Setting connection parameters
...
 
-.. _collocated_synapses
+.. _collocated_synapses:
 
 Collocated synapses
 ~~~~~~~~~~~~~~~~~~~
 It is now possible to create connections with several synapses simultaneously. The different synapse dictionaries will
-then be applied to each source-target pair. To create these collocated synapses, ``CollocatedSynapses()`` must be used
-as the `syn_spec` argument of ``Connect``, instead of the usual syn_spec dictionary argument. The constructor
+then be applied to each source-target pair. To create these collocated synapses, :py:func:`.CollocatedSynapses` must be used
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_handling_connections.rst:215]:
```suggestion
then be applied to each source-target pair. To create these collocated synapses, :py:class:`.CollocatedSynapses` must be used
```

**Review by steffengraber — APPROVED** [review]
:+1 from my side.

**Review by jasperalbers — APPROVED** [review]
Looks good to me!

**Comment by jougs** [comment]
@jessica-mitchell: I'll merge as soon as you fix the conflicts ;-)

**Comment by jougs** [comment]
@jessica-mitchell: can you maybe merge master again? If that does not solve the static code check problems, I can have a closer look.

**Comment by jougs** [comment]
I'm merging even thought the one static check still fails. I'll fix that in master.

**Pull request merged**